app/models/miq_schedule_worker/runner.rb

```diff
@@ -75,51 +75,44 @@ def schedules_for_all_roles
     schedule_category = :schedules_for_all_roles
 
     # Schedule - Log current system configuration
-    every = worker_settings[:log_active_configuration_interval]
-
     scheduler.schedule_every(
-      every,
-      :first_in => every,
+      worker_settings[:log_active_configuration_interval],
+      :first_in => :interval,
       :tags     => [:vmdb_appliance_log_config, schedule_category]
     ) { enqueue(:vmdb_appliance_log_config) }
 
     # Schedule - Log current database statistics and bloat
-    every = worker_settings[:log_database_statistics_interval]
     scheduler.schedule_every(
-      every,
-      :first_in => every,
```

Comment:
Isn't first_in => every the same as just not specifying a first_in?

I think it is... I believe we can drop first_in completely if it's the same as the interval.... less code

[first_in sets first_at](https://github.com/jmettraux/rufus-scheduler/blob/ba2b2d015c915be3146c304d70077222d7d70de4/lib/rufus/scheduler/jobs.rb#L423-L426)

[first_at is used if provided, otherwise uses the interval](https://github.com/jmettraux/rufus-scheduler/blob/ba2b2d015c915be3146c304d70077222d7d70de4/lib/rufus/scheduler/jobs.rb#L599-L608)

I'll test this but I'm pretty sure it's useless to specify first_in if it's the same as interval.

Reply:
```
irb(main):001:0> s = Rufus::Scheduler.new
=> #<Rufus::Scheduler:0x00007fe5778a8e50 @opts={}, @started_at=#<EtOrbi::EoTime:0x00007fe5778a8ae0 @seconds=1570814114.055407, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @paused=false, @jobs=#<Rufus::Scheduler::JobArray:0x00007fe5778a8d88 @mutex=#<Thread::Mutex:0x00007fe5778a8d60>, @array=[]>, @frequency=0.3, @mutexes={}, @work_queue=#<Thread::Queue:0x00007fe5778a8cc0>, @max_work_threads=28, @stderr=#<IO:<STDERR>>, @thread_key="rufus_scheduler_70311764903720", @scheduler_lock=#<Rufus::Scheduler::NullLock:0x00007fe5778a8ba8>, @trigger_lock=#<Rufus::Scheduler::NullLock:0x00007fe5778a8b80>, @thread=#<Thread:0x00007fe5778a8a90@/Users/joerafaniello/.gem/ruby/2.5.5/gems/rufus-scheduler-3.6.0/lib/rufus/scheduler.rb:545 sleep>>

irb(main):002:0> job = s.schedule_every(10.minutes, :first_in => 10.minutes) { puts 'hi'}
=> #<Rufus::Scheduler::EveryJob:0x00007fe577870de8 @scheduler=#<Rufus::Scheduler:0x00007fe5778a8e50 @opts={}, @started_at=#<EtOrbi::EoTime:0x00007fe5778a8ae0 @seconds=1570814114.055407, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @paused=false, @jobs=#<Rufus::Scheduler::JobArray:0x00007fe5778a8d88 @mutex=#<Thread::Mutex:0x00007fe5778a8d60>, @array=[#<Rufus::Scheduler::EveryJob:0x00007fe577870de8 ...>]>, @frequency=0.3, @mutexes={}, @work_queue=#<Thread::Queue:0x00007fe5778a8cc0>, @max_work_threads=28, @stderr=#<IO:<STDERR>>, @thread_key="rufus_scheduler_70311764903720", @scheduler_lock=#<Rufus::Scheduler::NullLock:0x00007fe5778a8ba8>, @trigger_lock=#<Rufus::Scheduler::NullLock:0x00007fe5778a8b80>, @thread=#<Thread:0x00007fe5778a8a90@/Users/joerafaniello/.gem/ruby/2.5.5/gems/rufus-scheduler-3.6.0/lib/rufus/scheduler.rb:545 sleep>>, @original=10 minutes, @opts={:first_in=>10 minutes}, @handler=#<Proc:0x00007fe577870f50@(irb):2>, @callable=#<Proc:0x00007fe577870f50@(irb):2>, @scheduled_at=#<EtOrbi::EoTime:0x00007fe577870d98 @seconds=1570814126.985564, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @unscheduled_at=nil, @last_time=nil, @locals={}, @local_mutex=#<Thread::Mutex:0x00007fe577870c80>, @id="every_1570814126.985564_70311764788980", @tags=[], @count=0, @last_work_time=0.0, @mean_work_time=0.0, @paused_at=nil, @times=nil, @first_at=#<EtOrbi::EoTime:0x00007fe577860b78 @seconds=1570814726.987073, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @next_time=#<EtOrbi::EoTime:0x00007fe577860b78 @seconds=1570814726.987073, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @last_at=nil, @frequency=10 minutes>

irb(main):004:0> job.next_time.to_time_s
=> "17:25:26.987072"

irb(main):006:0> job.first_at.to_time_s
=> "17:25:26.987072"
```

```
irb(main):001:0> s = Rufus::Scheduler.new
=> #<Rufus::Scheduler:0x00007ffe2e513f18 @opts={}, @started_at=#<EtOrbi::EoTime:0x00007ffe2e513900 @seconds=1570814274.907996, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @paused=false, @jobs=#<Rufus::Scheduler::JobArray:0x00007ffe2e513e50 @mutex=#<Thread::Mutex:0x00007ffe2e513e28>, @array=[]>, @frequency=0.3, @mutexes={}, @work_queue=#<Thread::Queue:0x00007ffe2e513c98>, @max_work_threads=28, @stderr=#<IO:<STDERR>>, @thread_key="rufus_scheduler_70364837748620", @scheduler_lock=#<Rufus::Scheduler::NullLock:0x00007ffe2e5139c8>, @trigger_lock=#<Rufus::Scheduler::NullLock:0x00007ffe2e513950>, @thread=#<Thread:0x00007ffe2e513860@/Users/joerafaniello/.gem/ruby/2.5.5/gems/rufus-scheduler-3.6.0/lib/rufus/scheduler.rb:545 sleep>>
irb(main):002:0> job = s.schedule_every(10.minutes) { puts 'hi'}
=> #<Rufus::Scheduler::EveryJob:0x00007ffe2e4d0b78 @scheduler=#<Rufus::Scheduler:0x00007ffe2e513f18 @opts={}, @started_at=#<EtOrbi::EoTime:0x00007ffe2e513900 @seconds=1570814274.907996, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @paused=false, @jobs=#<Rufus::Scheduler::JobArray:0x00007ffe2e513e50 @mutex=#<Thread::Mutex:0x00007ffe2e513e28>, @array=[#<Rufus::Scheduler::EveryJob:0x00007ffe2e4d0b78 ...>]>, @frequency=0.3, @mutexes={}, @work_queue=#<Thread::Queue:0x00007ffe2e513c98>, @max_work_threads=28, @stderr=#<IO:<STDERR>>, @thread_key="rufus_scheduler_70364837748620", @scheduler_lock=#<Rufus::Scheduler::NullLock:0x00007ffe2e5139c8>, @trigger_lock=#<Rufus::Scheduler::NullLock:0x00007ffe2e513950>, @thread=#<Thread:0x00007ffe2e513860@/Users/joerafaniello/.gem/ruby/2.5.5/gems/rufus-scheduler-3.6.0/lib/rufus/scheduler.rb:545 sleep>>, @original=10 minutes, @opts={}, @handler=#<Proc:0x00007ffe2e4d0c40@(irb):2>, @callable=#<Proc:0x00007ffe2e4d0c40@(irb):2>, @scheduled_at=#<EtOrbi::EoTime:0x00007ffe2e4d0b28 @seconds=1570814298.044724, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @unscheduled_at=nil, @last_time=nil, @locals={}, @local_mutex=#<Thread::Mutex:0x00007ffe2e4d0a60>, @id="every_1570814298.044724_70364837610940", @tags=[], @count=0, @last_work_time=0.0, @mean_work_time=0.0, @paused_at=nil, @times=nil, @first_at=nil, @next_time=#<EtOrbi::EoTime:0x00007ffe2e4d0128 @seconds=1570814898.04481, @zone=#<TZInfo::DataTimezone: Etc/UTC>, @time=nil>, @last_at=nil, @frequency=10 minutes>
irb(main):003:0> job.next_time.to_time_s
=> "17:28:18.044810"
```

Reply:
they look the same to me ^

Reply:
tldr; remove all first_in where first_in == interval for "every" jobs